로그인 및 회원가입 페이지 폼 구현

chaemina opened the pull request:
## Summary

Login 페이지를 구성하는 컴포넌트 및 폼을 구현함 


## Details

<img width="957" alt="image" src="https://github.com/Step3-kakao-tech-campus/Team18_FE/assets/103165845/91bae513-17d8-445d-801a-56e2654892ca">

<img width="959" alt="image" src="https://github.com/Step3-kakao-tech-campus/Team18_FE/assets/103165845/378d00fe-fdc6-48e0-acf6-30a33ecc5c87">

</br>
</br>

**Layout**
- 기존에 footer 컴포넌트를 템플릿에 호출하여 사용했으나, 뷰 포트 높이가 늘어남에 따라 하단에 고정이 어려워서 AccountLayout을 생성하여 레이아웃으로 관리되도록 수정함  

**로그인 페이지**
- 피그마 상에서 기획한 디자인은 왼쪽에 이미지와 같은 여백이 있었는데, 해당 부분에 삽입 할 이미지가 없다고 판단하여 요소들을 가운데 정렬 시킴 
- 컴포넌트는 Template에서 form으로 상수화 시킨 Login props를 전달하도록 함 

**회원가입 페이지**
- 로그인 페이지와 같이 입력창은 Inputbox에 props를 전달하도록 구현함
- 국가 선택의 경우, COUNTRY.js 상수 컴포넌트에 국가 객체 배열을 작성하여 이를 dropdown option의 prop으로 전달함
- 아직 멘토 선택 체크박스는 미구현 상태 


###  이슈번호 TO-DO

- [x]  account 관련 필요 컴포넌트 구현
- [x]  로그인 폼 구현 
- [x]  회원가입 폼 구현 



## Issue

- Related #14

chaemina committed:
fix : tailwind css color 수정 및 추가

title color 수정, paragraph color 추가

chaemina committed:
Merge branch 'weekly' of https://github.com/Step3-kakao-tech-campus/Team18_FE into feature/1-branch-account

chaemina committed:
Merge pull request #1 from Step3-kakao-tech-campus/weekly

레이아웃 구현

chaemina committed:
fix : xl size 추가

Button 컴포넌트 스타일에 xl size 추가

chaemina committed:
feat : Login page 폼 구현

Login form 에서 InputBox로 props를 전달하여, 컴포넌트의 재사용성을 높임
input의 속성 값들은 변동이 없는 값이므로LOGIN.js에 작성하여 따로
분리시켜 코드의 가독성을 높임

chaemina committed:
Merge branch 'feature/1-branch-account' of https://github.com/chaemina/Team18_FE into feature/1-branch-account

chaemina committed:
Merge branch 'weekly' of https://github.com/Step3-kakao-tech-campus/Team18_FE into feature/1-branch-account

chaemina committed:
fix : useEffect 두번 발생 방지

AuthCheck에서 effect가 두번 발생함을 방지하고자, useAuthCheck 를
작성하여 hook으로 사용함

chaemina committed:
refactoring : Link component로 수정

a 태그는 클릭 시 새로운 페이지를 불러오기 때문에  Link component로
수정함

jsh1147 commented on src/hooks/useAuthCheck.js:
```diff
@@ -0,0 +1,32 @@
+import { useRef, useState, useEffect } from "react";
+
+export const useAuthCheck = (effect) => {
```
AuthCheck 레이아웃에서 사용되는 훅이지만, 명칭 자체는 기능을 명시해주는 방향으로 가는게 더 좋을 것 같습니다! 전에 올려주셨던 참고 사이트처럼 useEffectOnce 이런 식이면 코드를 보는 사람이 해당 커스텀 훅을 이해하는데 도움이 될 것 같아요.

jsh1147 reviewed (approved):
수고하셨습니다!

chaemina committed:
refactoring : AccountLayout 추가

기존에 account template에서 footer 컴포넌트를 호출하여 사용했는데, 이는
요소가 넘치는 경우에 하단에 위치시키기 어려움이 있어서 accountlayout을
새로 작성하여 route로  레이아웃 적용시키도록 수정함

chaemina committed:
fix : overflow 처리

dropdown 컴포넌트의 옵션이 많아 질 경우, 컴포넌트가 스크린 범위 이상으로
커짐을 방지하고자 max-height 값을 설정하고, 제한 범위 이상일 때 y 스크롤
생성하도록 수정함

chaemina committed:
feat : 회원가입 컴포넌트 및 폼 구현

Login 폼과 동일하게 입력란은 inputBox에 props를 전달하도록 구현함
국가 선택의 경우, COUNTRY.js 상수 컴포넌트를 생성하여 국가 이름 배열을
저장하고 dropdown option의 prop으로 전달함
*멘토 선택 미 구현
